File: Home2WorkAPI/src/main/java/it/gruppoinfor/home2workapi/model/Company.java
package it.gruppoinfor.home2workapi.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Company implements Serializable {

    @SerializedName("Id")
    @Expose
    private Long id;
    @SerializedName("Name")
    @Expose
    private String name;
    @SerializedName("LatLng")
    @Expose
    private LatLng location;
    @SerializedName("Address")
    @Expose
    private Address address;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LatLng getLocation() {
        return location;
    }

    public void setLocation(LatLng location) {
        this.location = location;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    @Override
    public String toString() {
        return name + " (" + address.getCity() + ")";
    }
}

File: Home2WorkAPI/src/main/java/it/gruppoinfor/home2workapi/model/Company.kt
package it.gruppoinfor.home2workapi.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

class Company : Serializable {

    @SerializedName("Id")
    @Expose
    var id: Long = 0
    @SerializedName("Name")
    @Expose
    var name: String = ""
    @SerializedName("LatLng")
    @Expose
    var location: LatLng = LatLng()
    @SerializedName("Address")
    @Expose
    var address: Address = Address()

    override fun toString(): String {
        return "$name (${address.city})"
    }
}